app/src/main/java/fiware/smartcity/marketplace/MarketActivity.java

```diff
@@ -31,38 +32,51 @@
     private static Activity activity;
 
     private WebView webView;
-    private ProgressBar mPbar;
+    private RelativeLayout mPbar;
 
     private class SpinnerClient extends WebViewClient {
         @Override
         public void onPageFinished(WebView view, String url) {
+            view.setVisibility(View.VISIBLE);
             mPbar.setVisibility(View.GONE);
-            webView.setVisibility(View.VISIBLE);
+        }
+
+        @Override
+        public void onPageStarted(WebView view, String url, Bitmap favicon) {
+            mPbar.setVisibility(View.VISIBLE);
         }
     }
 
     private class WebAppTokenInterface {
-        Context mContext;
+        SharedPreferences.Editor prefsEditor;
 
         WebAppTokenInterface(Context c) {
-            mContext = c;
+            SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(c);
```

Comment:
instead of putting this code on the constructor, which it is weird, just create and auxiliary function called 'openPreferences()' which could be used by both 'save' and 'clear' token.